great_expectations/expectations/expectation.py

```diff
@@ -2373,6 +2369,12 @@ class BatchExpectation(Expectation, ABC):
     domain_type: ClassVar[MetricDomainTypes] = MetricDomainTypes.TABLE
     args_keys: ClassVar[Tuple[str, ...]] = ()
 
+    @pydantic.validator("mostly")
+    def validate_mostly(mostly: float) -> float:
+        if not (0 <= mostly <= 1):
+            raise ValueError("mostly must be a value between 0 and 1")
+        return mostly
```

Comment:
Originally just used `Field` with `ge` and `le` but that didn't seem to mesh well with inheritance?

This validator works with child classes but I saw test failures with the other pattern - worth investigating further.